### PR TITLE
domain white listing added for api auth

### PR DESCRIPTION
**Pull request by rex-god**

### CODE REVIEW

File: src/app.service.ts [hunk 1]
```diff
@@ -45,4 +46,14 @@ export class AppService {
       throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
     }
   }
+
+  async whiteListDomains(dto: WhiteListDomainsDto): Promise<User> {
```
Comment (vgvishesh) [inline src/app.service.ts:50]: What if a user wants to just add 1 more endpoint to an already existing list of whitelisted domains?

Comment (vgvishesh) [reply src/app.service.ts:50]: This approach will not work, as the new one will overwrite the old string[]

Comment (rex-god) [reply src/app.service.ts:50]: The new approach mentioned at Notion.

File: src/app.service.ts [hunk 2]
```diff
@@ -45,4 +46,40 @@ export class AppService {
       throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
     }
   }
+
+  async whiteListDomain(dto: WhiteListDomainsDto): Promise<User> {
+    try {
+      const { api_key, domain } = dto;
+      const domainsObj = await this.getWhiteListDomains(api_key);
```
Comment (vgvishesh) [inline src/app.service.ts:53]: This is not a good way to update, for multiple reasons, but importantly this is slow. 
Can you update the object directly in DB without fetching it in the application and then putting it back?

File: src/app.service.ts [hunk 3]
```diff
@@ -45,4 +46,40 @@ export class AppService {
...
+      return result;
+    } catch (error) {
+      throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+    }
+  }
+
+  async deleteDomainFromWhiteList(dto: WhiteListDomainsDto): Promise<any> {
+    try {
+      const { api_key, domain } = dto;
+      const domainsObj = await this.getWhiteListDomains(api_key);
```
Comment (vgvishesh) [inline src/app.service.ts:65]: For a similar reason as above, this is a slow way to do it. Please see how can we do this without fetching the object in the application first.

**Pull request merged**